### PR TITLE
chore(repo): ESLint refactor + speed up 70%

### PR DESCRIPTION
**Pull request by Nodonisko**
I found our ESLint setup to be too slow, so I refactored and cleaned it up to increase its speed by approximately 70%.

To start, we should revisit the fundamental purpose of linters. Linters are meant to perform static analysis on code, typically on a single file basis, by parsing it into an AST (Abstract Syntax Tree) and then applying linting rules. Anything beyond this scope isn't true linting and can significantly slow down the process. For instance, integrating Prettier into ESLint is unnecessary since we have a separate formatting check and most editors support auto-formatting with Prettier upon saving. You can use this plugin for Visual Studio Code: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode (similar plugins exist for other IDEs).

Certain plugins and rules, especially those from `eslint-plugin-import` like `import/no-cycle`, are highly computational because they analyze the entire dependency tree, which slows down the process further. This is particularly true for `.ts` files as it involves TypeScript specifics. Moreover, the `import/no-cycle` rule is not reliable, given the numerous cyclic dependencies in our desktop suite, which are already detected by Nx and TypeScript.

These factors accounted for over 50% of the linter's runtime. With a few additional tweaks, I managed to reduce the linting time for the entire repository to about 35 seconds. This improvement makes it feasible to always lint the entire repo rather than just specific packages. I plan to test this approach on our CI to see if it's faster than using Nx for linting. If so, we may adopt this method for CI as Nx tends to be slower for full-repo runs, though faster for smaller, package-specific runs.

**Linter Runtime Before Changes: 1m 30s ~ 2 min**

Before Changes Screenshot

**Linter Runtime After Changes: ~30s**

After Changes Screenshot

Summary of changes made:

1. Disabled Prettier in ESLint.
2. Disabled the `import/no-cycle` rule.
3. Removed the `airbnb-eslint-preset` as it was unnecessarily strict. I opted for some default recommended presets instead.
4. Replaced the Nx linting process with a direct ESLint run across the entire repo. I also optimized the GitHub Actions cache to speed up subsequent runs (to a maximum of 5s).
5. Reconfigured some rules to better meet our needs and disabled others that were frequently bypassed in our code.

// edit: disabled few more rules that were not used anyway and it's even faster now

### CODE REVIEW

**Comment by Nodonisko** [comment]
@SocketSecurity ignore npm/@nrwl/tao@18.0.4

File: .eslintrc.js [hunk 1]
```diff
@@ -113,7 +114,8 @@ module.exports = {
                 },
             },
         ],
-        'import/no-cycle': 'error',
+        // Could be useful, but it's very very very slow
+        'import/no-cycle': 'off',
```
Comment (juriczech) [inline .eslintrc.js:118]: Not sure this is wanted

Comment (Nodonisko) [reply .eslintrc.js:118]: Explain in PR description

Comment (juriczech) [reply .eslintrc.js:118]: My bad. I wrote the comment before and didn't notice description was updated.

**Review by komret — APPROVED** [review]
❤️ 👏

**Pull request merged**